src/ATen/native/xpu/AdaptiveAveragePooling2d.cpp

```diff
@@ -72,11 +72,20 @@ Tensor mean_backward(
     c10::SymIntArrayRef shape,
     OptionalIntArrayRef opt_dim,
     c10::SymInt numel,
-    bool keepdim) {
+    bool keepdim,
+    const Tensor& input) {
   bool is_all_reduce = !opt_dim.has_value() || opt_dim.value().empty();
   auto n =
       is_all_reduce ? std::move(numel) : _safe_size(shape, opt_dim.value());
-  return sum_backward(grad, shape, opt_dim, keepdim) / std::move(n);
+
+  Tensor grad_input_ =
+      sum_backward(grad, shape, opt_dim, keepdim) / std::move(n);
+
+  if (input.suggest_memory_format() == at::MemoryFormat::ChannelsLast) {
+    grad_input_ = grad_input_.contiguous(input.suggest_memory_format());
```

Comment:
I could not understand the acc failure. The fixing here adjusts the memory format, there is no change in content of `grad_input_`. So when the next node in the autograd graph could handle plain format, the model should work.

Reply:
Synced offline. There is an assumption in Inductor usage with Triton kernel. Inductor generates hardcoded memory access for specific memory layout. For example, Inductor enables CL optimization for a model, the Triton kernel will be generated for CL only. The side-effect is fallback operators should propagate memory layout correctly.